src/corehost/common/pal.unix.cpp

```diff
@@ -357,23 +359,35 @@ pal::string_t pal::get_current_os_rid_platform()
             // Loop until we are at the end of file
             while (!fsVersionFile.eof())
             {
-                size_t pos = line.find(strID);
-                if ((pos != std::string::npos) && (pos == 0))
+                // Look for ID if we have not found it already
+                if (!fFoundID)
                 {
-                    ridOS.append(line.substr(3));
-                    ridOS.append(_X("."));
-                    fFoundID = true;
+                    size_t pos = line.find(strID);
+                    if ((pos != std::string::npos) && (pos == 0))
+                    {
+                        valID.append(line.substr(3));
+                        fFoundID = true;
+                    }
                 }
 
-                pos = line.find(strVersionID);
-                if ((pos != std::string::npos) && (pos == 0))
+                // Look for VersionID if we have not found it already
+                if (!fFoundVersion)
                 {
-                    ridOS.append(line.substr(11));
-                    fFoundVersion = true;
+                    size_t pos = line.find(strVersionID);
+                    if ((pos != std::string::npos) && (pos == 0))
+                    {
+                        valVersionID.append(line.substr(11));
+                        fFoundVersion = true;
+                    }
                 }
 
                 if (fFoundID && fFoundVersion)
```

Comment:
Sure - but I am not sure we want to maintain consistency in missing data. Are you suggesting we do the same (no place anything if either one is not found) in managed code as well?